cmd/readiness/main.go

```diff
@@ -221,7 +221,16 @@ func main() {
 		panic(err)
 	}
 
-	cfg, err := config.BuildFromEnvVariables(clientSet, isHeadlessMode())
+	healthStatusFilePath := config.GetEnvOrDefault(config.AgentHealthStatusFilePathEnv, config.DefaultAgentHealthStatusFilePath)
+	file, err := os.Open(healthStatusFilePath)
+	// The agent might be slow in creating the health status file.
+	// In that case, we don't want to panic to show the message
+	// in the kubernetes description. That would be a red herring, since that will solve itself with enough time.
+	if err != nil {
+		os.Exit(1)
```

Comment:
can we log something here to indicate what happened?

Reply:
I was thinking about the same, but then I would need to move the logger initialization out to fit the logger: https://github.com/mongodb/mongodb-kubernetes-operator/blob/a7812ff0728930f2fb35ae4a9776b80a86e2e64b/pkg/readiness/config/config.go#L59-L64

I **could** do this 🤔

Reply:
How about panicing with an error? This way it should be visible in the `kubectl get events` and `kubectl describe pod`.

Reply:
Thanks for the explanation @nammn ! I believe it's still worth logging an info why this happened. Without it, investigating corner cases might be simply impossible.

Reply:
Done